### PR TITLE
added the updates to indexing that D&D requested

### PR DESCRIPTION
**Pull request by niquerio**
Added some new fields that were requested by D&D. Also moved some of the logic currently in spectrum here.

All of the changes here are in my `docker` branch in `catalog_solr_index`. (`catalog_solr_index` isn't set up on the production machines so that's why changes are still being made here.)

There's a corresponding PR in library_solr_config https://github.com/mlibrary/library_solr_config/pull/20 

I'd like feedback on `contributors` and `related_title`. The logic was complicated enough that I did it this way, but if there's a more succinct way to do it I'm open.

### CODE REVIEW

File: indexers/common.rb [hunk 1]
```diff
@@ -174,6 +174,68 @@ def record_is_umich(r, context)
...
 end
 
+#changes by mrio Feb 2022
+to_field "main_author_display", extract_marc("100abcdefgjklnpqtu4:101abcdefgjklnpqtu4:110abcdefgjklnpqtu4:111abcdefgjklnpqtu4")
+to_field "main_author", extract_marc("100abcdgjkqu:101abcdgjkqu:110abcdgjkqu:111abcdgjkqu")
+
+def concat_subfields(subfields, f)
+  subfields.split('').map{|x| f[x]}.compact.join(" ")
+end
+to_field "contributors_display" do |rec, acc|
```
Comment (billdueber) [inline indexers/common.rb:184]: You could probably leverage [`extract_marc_unless`](https://github.com/mlibrary/umich_catalog_indexing/blob/a910dae6eaf10b15e987b6926033959157a24d83/lib/ht_traject/ht_macros.rb#L8-L79) here. It's the same as `extract_marc`, but takes a lambda of the form `my_skip_function(rec, field) -> boolean` -- the two arguments are the whole record and the specific field under consideration. You could use a lambda like `skipif = ->(rec, field) { field.indicator2 == "2" && f["t"].nil? }` and not have to go through all this messing around.

Comment (billdueber) [reply indexers/common.rb:184]: You can see [an example of its use](https://github.com/mlibrary/umich_catalog_indexing/blob/ccd6cd170c26ce3a1bd4310ada99ae3d0ca22b31/indexers/common.rb#L147-L157) where I had to ignore JSTOR claiming authorship.

Comment (billdueber) [reply indexers/common.rb:184]: And given that you're also messing with the 7xx author fields, you'll want to incorporate the `skipWaSeSS` check as well.

File: indexers/common.rb [hunk 2]
```diff
@@ -174,6 +174,68 @@ def record_is_umich(r, context)
...
+  end
+end
+
+to_field "related_title_display" do |rec, acc|
+  rec.each_by_tag(["730"]) do |f|
+    if f.indicator2 == " "
+      acc << concat_subfields("abcdefghjklmnopqrstuvwxyz",f)
+    end
+  end
+  ["700","710","711"].each do |tag|
```
Comment (billdueber) [inline indexers/common.rb:210]: Are they sure they want these 7xx personal/corporate names in the title display? Just making sure it's not a cut-and-paste error.

File: indexers/common.rb (hunk 1 above)
Comment (niquerio) [reply indexers/common.rb]: Done! I knew there was a better way to do this.

File: indexers/common.rb (hunk 2 above)
Comment (niquerio) [reply indexers/common.rb]: This is all from this doc https://docs.google.com/spreadsheets/d/1Ic_S3LrzNy5gwpXX8VDA2GoTJ44ELt2mGJamO_LsBvM/edit#gid=0

I'm fairly certain that this is what's wanted.

**Pull request merged**